py/vm.c

```diff
@@ -1462,6 +1462,7 @@ unwind_jump:;
             // - constant GeneratorExit object, because it's const
             // - exceptions re-raised by END_FINALLY
             // - exceptions re-raised explicitly by "raise"
+            assert(nlr.ret_val == &mp_const_GeneratorExit_obj || code_state->ip);
```

Comment:
I'm not sure why this is needed.  `code_state->ip` is never NULL in this entire function, so I wonder why the compiler thinks it might be?

If this is still necessary, it should probable be `assert(code_state->ip != NULL)` and be put at the very top of this function.